### ISSUE TITLE
Jooq 3.11.9 fails with MySQL 8

### ISSUE THREAD

**Issue by harshit29dx** [jOOQ/jOOQ#8269]
### Versions:

- jOOQ: 3.11.9
- Java:  8
- Database (include vendor): MySQL 8
- OS: Mac
- JDBC Driver (include name if inofficial driver): mysql-connector-java 5.1.44

I get the following error:

```
 org.jooq.exception.DataAccessException: SQL [select 1 as `one` from dual where exists (select 1 as `one` from `mysql`.`proc`)]; Table 'mysql.proc' doesn't exist
	at org.jooq_3.11.9.MYSQL.debug(Unknown Source) ~[?:?]
	at org.jooq.impl.Tools.translate(Tools.java:2384) ~[jooq-3.11.9.jar:?]
	at org.jooq.impl.DefaultExecuteContext.sqlException(DefaultExecuteContext.java:832) ~[jooq-3.11.9.jar:?]
	at org.jooq.impl.AbstractQuery.execute(AbstractQuery.java:364) [jooq-3.11.9.jar:?]
	at org.jooq.impl.AbstractResultQuery.fetchLazy(AbstractResultQuery.java:393) [jooq-3.11.9.jar:?]
	at org.jooq.impl.AbstractResultQuery.fetchLazy(AbstractResultQuery.java:380) [jooq-3.11.9.jar:?]
	at org.jooq.impl.AbstractResultQuery.fetchOne(AbstractResultQuery.java:545) [jooq-3.11.9.jar:?]
	at org.jooq.impl.SelectImpl.fetchOne(SelectImpl.java:2879) [jooq-3.11.9.jar:?]
	at org.jooq.impl.DefaultDSLContext.fetchExists(DefaultDSLContext.java:4345) [jooq-3.11.9.jar:?]
	at org.jooq.impl.DefaultDSLContext.fetchExists(DefaultDSLContext.java:4350) [jooq-3.11.9.jar:?]
	at org.jooq.meta.mysql.MySQLDatabase.is8(MySQLDatabase.java:233) [jooq-meta-3.11.9.jar:?]
	at org.jooq.meta.mysql.MySQLDatabase.getRoutines0(MySQLDatabase.java:467) [jooq-meta-3.11.9.jar:?]
	at org.jooq.meta.AbstractDatabase.getRoutines(AbstractDatabase.java:1664) [jooq-meta-3.11.9.jar:?]
	at org.jooq.codegen.JavaGenerator.generate(JavaGenerator.java:571) [jooq-codegen-3.11.9.jar:?]
	at org.jooq.codegen.JavaGenerator.generate(JavaGenerator.java:458) [jooq-codegen-3.11.9.jar:?]
	at org.jooq.codegen.JavaGenerator.generate(JavaGenerator.java:380) [jooq-codegen-3.11.9.jar:?]
	at org.jooq.codegen.GenerationTool.run(GenerationTool.java:752) [jooq-codegen-3.11.9.jar:?]
	at org.jooq.codegen.GenerationTool.generate(GenerationTool.java:222) [jooq-codegen-3.11.9.jar:?]
	at com.ovo.paas.jooqdb.startup.JooqStartup.init(JooqStartup.java:50) [classes/:?]
	at com.ovo.paas.jooqdb.startup.JooqStartup.main(JooqStartup.java:58) [classes/:?]
Caused by: com.mysql.jdbc.exceptions.jdbc4.MySQLSyntaxErrorException: Table 'mysql.proc' doesn't exist
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method) ~[?:1.8.0_181]
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62) ~[?:1.8.0_181]
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45) ~[?:1.8.0_181]
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423) ~[?:1.8.0_181]
	at com.mysql.jdbc.Util.handleNewInstance(Util.java:425) ~[mysql-connector-java-5.1.44.jar:5.1.44]
	at com.mysql.jdbc.Util.getInstance(Util.java:408) ~[mysql-connector-java-5.1.44.jar:5.1.44]
	at com.mysql.jdbc.SQLError.createSQLException(SQLError.java:943) ~[mysql-connector-java-5.1.44.jar:5.1.44]
	at com.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:3973) ~[mysql-connector-java-5.1.44.jar:5.1.44]
	at com.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:3909) ~[mysql-connector-java-5.1.44.jar:5.1.44]
	at com.mysql.jdbc.MysqlIO.sendCommand(MysqlIO.java:2527) ~[mysql-connector-java-5.1.44.jar:5.1.44]
	at com.mysql.jdbc.MysqlIO.sqlQueryDirect(MysqlIO.java:2680) ~[mysql-connector-java-5.1.44.jar:5.1.44]
	at com.mysql.jdbc.ConnectionImpl.execSQL(ConnectionImpl.java:2487) ~[mysql-connector-java-5.1.44.jar:5.1.44]
	at com.mysql.jdbc.PreparedStatement.executeInternal(PreparedStatement.java:1858) ~[mysql-connector-java-5.1.44.jar:5.1.44]
	at com.mysql.jdbc.PreparedStatement.execute(PreparedStatement.java:1197) ~[mysql-connector-java-5.1.44.jar:5.1.44]
	at org.jooq.tools.jdbc.DefaultPreparedStatement.execute(DefaultPreparedStatement.java:209) ~[jooq-3.11.9.jar:?]
	at org.jooq.impl.Tools.executeStatementAndGetFirstResultSet(Tools.java:3494) ~[jooq-3.11.9.jar:?]
	at org.jooq.impl.AbstractResultQuery.execute(AbstractResultQuery.java:268) ~[jooq-3.11.9.jar:?]
	at org.jooq.impl.AbstractQuery.execute(AbstractQuery.java:350) ~[jooq-3.11.9.jar:?]
	... 16 more
```

**Comment by lukaseder** [comment]
Thank you very much for your report and for linking it from https://stackoverflow.com/q/54438532/521799. As mentioned in that stack overflow question: 

> That query is there precisely to check whether you're running on MySQL 8+. It should not cause an error or even a stack trace (but maybe a debug message). You can safely ignore it.

This exception should not cause any issues to the code generation process. It should not even be logged. Are you sure this problem is causing failure to your code generation? If so, can you please post a bit more information on how to reproduce this issue? In particular, what's `JooqStartup` doing?

**Comment by harshit29dx** [comment]
@lukaseder so actually the generation was throwing an Exception which was caught by the code and printed. Code Generation, however was successful. My bad.

**Comment by lukaseder** [comment]
@harshit29dx : But who is logging the exception? Do you have a logger / execute listener turned on?

**Comment by lukaseder** [comment]
... I'm asking because this exception should be "silent" and not leave any traces as it would confuse users. As could be seen :)

**Comment by lukaseder** [comment]
I cannot reproduce this. Without further reproduction instructions, I'm closing this as "worksforme"

**Comment by amin3mej** [comment]
@lukaseder Could you fix this problem? How?

**Comment by lukaseder** [comment]
@amin3mej This problem could not be reproduced. If you're having a similar problem, please create a new issue, preferably with instructions on how to reproduce it.

**Comment by amin3mej** [comment]
@lukaseder Thanks.
 For anyone who has the same problem: I solved it by downgrading the version from 3.13.1 to 3.12.4 and the problem got fixed.
